src/local_insight_engine/main.py

```
@@ -46,66 +68,178 @@ def analyze_document(self, document_path: Path, factual_mode: bool = False) -> d
+    def _persist_analysis(self, document_path: Path, processed_data, analysis: dict, factual_mode: bool):
+        """Persist analysis results to database for future Q&A sessions"""
+        try:
+            from .persistence.models import PersistentQASession, QAExchange
+            from datetime import datetime
+            import json
+
+            logger.database_operation("Persisting analysis results")
+
+            with self.db_manager.get_session() as session:
+                # Create or update document record
+                doc_record = session.query(Document).filter_by(
```

Ruff (0.12.2)

Undefined name `Document`

(F821)